frontend/app/services/address-service.server.ts

```diff
@@ -16,22 +16,22 @@ const addressDtoSchema = z.object({
 type AddressDto = z.infer<typeof addressDtoSchema>;
 
 const addressInfoSchema = z.object({
-  address: z.string().optional(),
-  city: z.string().optional(),
+  address: z.string(),
+  city: z.string(),
   province: z.string().optional(),
   postalCode: z.string().optional(),
-  country: z.string().optional(),
+  country: z.string(),
 });
 
 export type AddressInfo = z.infer<typeof addressInfoSchema>;
 
 function toAddressInfo(addressDto: AddressDto): AddressInfo {
   return {
-    address: addressDto.addressApartmentUnitNumber ? addressDto.addressApartmentUnitNumber + ' ' + addressDto.addressStreet : addressDto.addressStreet,
-    city: addressDto.addressCity,
+    address: addressDto.addressApartmentUnitNumber ? addressDto.addressApartmentUnitNumber + ' ' + addressDto.addressStreet : addressDto.addressStreet ?? ' ',
+    city: addressDto.addressCity ?? ' ',
     province: addressDto.addressProvince,
     postalCode: addressDto.addressPostalZipCode,
-    country: addressDto.addressCountry,
+    country: addressDto.addressCountry ?? ' ',
```

Comment:
Rather than defaulting `city` and `country` to a `' '` value, we could make the assumption that the `addressDtoSchema` will have mandatory `addressCity` and `addressCountry` fields.